File: examples/swiftui/duis07_duplicate_view_bad.swift
import SwiftUI

/// DUIS-07 – DUPLICATE VIEW
///
/// Three user "cards" with nearly identical view structure are copy–pasted
/// instead of extracting a reusable UserCard view.

struct DuplicateViewExampleView: View {
    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            Text("Duplicate View Example")
                .font(.headline)

            // Card #1
            VStack(alignment: .leading) {
                Text("Alice").bold()
                Text("alice@example.com")
                    .foregroundColor(.secondary)
            }
            .padding()
            .background(Color(white: 0.95))
            .cornerRadius(8)

            // Card #2 – copy-pasted layout
            VStack(alignment: .leading) {
                Text("Bob").bold()
                Text("bob@example.com")
                    .foregroundColor(.secondary)
            }
            .padding()
            .background(Color(white: 0.95))
            .cornerRadius(8)

            // Card #3 – copy-pasted again
            VStack(alignment: .leading) {
                Text("Charlie").bold()
                Text("charlie@example.com")
                    .foregroundColor(.secondary)
            }
            .padding()
            .background(Color(white: 0.95))
            .cornerRadius(8)

            Spacer()
        }
        .padding()
    }
}

File: examples/compose/duis07_duplicate_view_bad.kt
package examples.compose

import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

/**
 * DUIS-07 – DUPLICATE VIEW
 *
 * Three user "cards" with the same layout are copy–pasted
 * instead of extracting a reusable composable.
 */

@Composable
fun DuplicateViewExample() {
    Scaffold(
        topBar = { TopAppBar(title = { Text("Duplicate View Example") }) }
    ) { padding ->
        Column(
            modifier = Modifier
                .padding(padding)
                .padding(16.dp)
        ) {
            // Card #1
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(bottom = 8.dp)
            ) {
                Column(Modifier.padding(8.dp)) {
                    Text("Alice", style = MaterialTheme.typography.titleMedium)
                    Text("alice@example.com")
                }
            }

            // Card #2 – copy-pasted layout
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(bottom = 8.dp)
            ) {
                Column(Modifier.padding(8.dp)) {
                    Text("Bob", style = MaterialTheme.typography.titleMedium)
                    Text("bob@example.com")
                }
            }

            // Card #3 – again copy-pasted
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(bottom = 8.dp)
            ) {
                Column(Modifier.padding(8.dp)) {
                    Text("Charlie", style = MaterialTheme.typography.titleMedium)
                    Text("charlie@example.com")
                }
            }
        }
    }
}
